### DISCUSSION TITLE
Expectations on running with this repo

### DISCUSSION THREAD

**Discussion by markwigzell** [santeri3700/vive-pro-2-on-linux#2] [General]
First off, many thanks for the work! It is appreciated.
After purchasing the Vive Pro 2 recently, I have finally come down to earth and realized I "made a mistake" in that I assumed the Vive Pro support was sufficient. I guess was too naive. I normally run Steam with Proton and various Windows games such as "No Mans Sky" so I had the impression all would work well. 
My concrete issue is after connecting the equipment and powering on the link box, the HMD has a red light. and SteamVR reports a missing driver.
I am not normally a tinkerer but I have worked the kernel before, and I use ARCH today, so I feel I might be able to attempt this.
I understand the steam architecture so little, that :

- I have problems discerning from the readme on this repo whether or not Proton is supported?
- will this patch be applied by a pacman update to newer kernels or is this a manual affair?
- is it expected that the kernel will support the vive pro 2 eventually? (what is the time line for that?)

**Comment by santeri3700** [comment]
Hi, since you use Arch Linux, the amount of manual tinkering is quite small. As of now, the [kernel patches ](https://github.com/CertainLach/VivePro2-Linux-Driver/tree/master/kernel-patches)and [unofficial proxy driver](https://github.com/CertainLach/VivePro2-Linux-Driver) are required to make the headset work on Linux.

And to answer your other questions:
- Proton should work with the same level of compatibility as other officially supported SteamVR headsets. I have tested [Beat Saber](https://www.youtube.com/watch?v=hN59l0qfRTs) and VRChat myself. \
ProtonDB can also give you more information about the current state of VR compatibility: https://www.protondb.com/explore?selectedFilters=userTags&selectedTags=VR
- At the moment, installing/updating the patched kernel is manual work and requires pulling this repository. I'll try to keep the custom kernel version relatively up-to-date when I have time. It is also possible for you to manually modify the [PKGBUILD](https://github.com/santeri3700/vive-pro-2-on-linux/blob/master/PKGBUILD) to build newer kernels than what I've "provided", but the patches might not be compatible.
- The kernel probably will support the Vive Pro 2 out-of-the-box in the future. This is a question better directed at the proxy driver and patch developer [CertainLach](https://github.com/CertainLach/VivePro2-Linux-Driver). I've merely just written a guide and a custom kernel package for Arch based on their work. \
Note that it is not only the kernel support that is required, but also the SteamVR driver support (which I think Vive Pro 2 owners probably won't ever get officially)

If you want to have as smooth as possible VR experience on Linux, then I would recommend getting a Valve Index or the original HTC Vive. \
The current status of VR on Linux in general is a bit of an experiment to be honest. With Vive Pro 2, you just have to jump though a couple more hoops to get there. \
You can read more about VR on Linux here: https://gitlab.com/vr-on-linux/VR-on-Linux

**Reply by mwigzell** [reply]
Hi, I used your instructions to both create a patched low latency arch kernel and to install the drivers from CertainLach. I have had some success and also some issues I'm facing now. Firstly, the patched kernel froze when the systemd runlevel reached graphical.target. So I abandoned the kernel. Not sure what that is, I didn't investigate further. Secondly, wow, the installed drivers got me past my initial issue of "no driver". Steam VR opened up and prompted me to define the room etc. I am using Valve Base 2 stations, and it detected them both on channel 12. But I couldn't get the Steam VR menu to navigate to where I could resolve the channel collision. 
This issue was due in part at least, to the screen going black when I clicked on the UI. I had to click again on the window background to get the UI to show up. This issue persisted till I could close all the sub-dialogs. Then it started behaving, I can see from the logs that everything is being detected and SteamVR wants to update all the firmware, for the HMD, the 2 base stations and the 2 controllers. I am also able to see the initial "room" that it is trying to calibrate the HMD for. The picture has artifacts, different ones in each eye. 
So I am wondering how to proceed? Should I trust the drivers to update the firmware? If so, in what order should I update that firmware? 
Also, do you know why the UI is misbehaving so badly? It seems like the mouse clicks are being swallowed when trying to click on SteamVR UI panes.

**Reply by santeri3700** [reply]
I'm not sure what the problem is with the custom kernel and graphical.target, but I've been able to use X11 and Wayland.

Did you use your pre-existing kernel config or did you create a new one from scratch? Some of the kernel patches are required for the HMD to function properly.

Check the README of this repository for the picture artifact problem and workaround. You probably have to run SteamVR without Async reprojection enabled (it's broken on AMD/Mesa).

Firmware updates should work like they work on Windows and I don't think the order matters (ofc do this at your own risk, this is unofficial after all).

And for the SteamVR UI problems, I personally haven't had any of those issues. I'd suggest trying the SteamVR Beta and possibly other desktop environment. SteamVR might be buggy on Wayland or Tiling window managers in general.

**Reply by mwigzell** [reply]
Just to clarify: I'm using an Nvidia GeForce GTX 1660 with nvidia drivers. I didn't use the suggested SteamVR, I used latest SteamVR Beta. I'm not running Wayland. Regarding the kernel patches, I patched my existing kernel config, as suggested, copying it from /proc. I'll update the firmware and turn off Async reprojection and report back.

**Reply by mwigzell** [reply]
Hey, well I was able to update the firmware on the hmd and the controllers. But it failed to update the base stations, I think it was unable to consistently reach them. The issue is complicated by the fact that SteamVR first detects that the base stations conflict on channel 12, then declares that no base station is broadcasting when I click to resolve it. The UI still has this weird issue whereby mouse clicks are swallowed and it does nothing, or the click goes through and screen goes black, I have to click where there is some screen background to get it back. WT? 
Probably the most serious issue though is the video in the HMD is not horizontally synced properly. It has a weird kind of wrap around, and the right hand fifth of each eye has vertical lines of different colors. These do not show up in the VR monitor screen.
I played with showing the performance graph, and it is mostly a green band with an occasional short patch of orange. I think it is handling the latency. When I turn my head with the HMD on, the image is refreshed just fine. (but still the sync glitsch).
I should probably broaden this to a forum right? Any ideas where I might try to get some help?

**Reply by santeri3700** [reply]
I found something that could help you:
- Check this comment for a possible workaround to the graphics problems: https://github.com/CertainLach/VivePro2-Linux-Driver/issues/15#issuecomment-1369758657
- Check this Reddit thread for the Base station issues: https://www.reddit.com/r/vive_vr/comments/c8bil1/is_there_a_way_to_change_base_station_20_channels/ \
 
I guess you could also ask [CertainLach](https://github.com/CertainLach/VivePro2-Linux-Driver) about the custom driver and graphics related things.
Also, Nvidia compatibility is also kinda iffy since the drivers are proprietary and can't be easily modified unlike the AMD kernel drivers and Mesa.

You could also try the [SteamVR for Linux GitHub project](https://github.com/ValveSoftware/SteamVR-for-Linux/), [r/virtualreality_linux](https://www.reddit.com/r/virtualreality_linux/) or [SteamVR forums](https://steamcommunity.com/app/250820/discussions/5/) for the base station and UI issues. I wouldn't be surprised if other people have experienced and reported similar issues.

As my repository says, VR on Linux is at a pretty experimental and mediocre level at the moment, especially with unofficial drivers and kernel patches. We can only hope that Valve Deckard will improve the situation in the future.
